docs: supersede changelog ADR, add new ADR and update PR template

alvarothomas opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us process your pull request.

Please make sure to fulfil our contribution guidelines (https://developer.shopware.com/docs/resources/guidelines/code/contribution?category=shopware-platform-dev-en/contribution).

Do your changes need to be documented?
Please create a second pull request at https://github.com/shopware/docs
-->

### 1. Why is this change necessary?

The previous changelog ADR (2020-08-03-implement-new-changelog.md) and PR checklist no longer match our actual release documentation process.  
We are now using in-repo `RELEASE_INFO` and `UPGRADE` files as the source of truth for developer-facing release notes, with the full changelog being auto-generated from GitHub releases.

This update ensures that contributors and reviewers follow the new workflow and that the repository’s ADRs reflect the current process.

---

### 2. What does this change do, exactly?

- Moves the old ADR (`2020-08-03-implement-new-changelog.md`) to `adr/_superseded/`.  
- Adds a new ADR: `2025-10-28-changelog-release-info-process.md`, describing the curated in-repo release documentation process (`RELEASE_INFO-6.x.md` and `UPGRADE-6.x.md`).  
- Updates `.github/PULL_REQUEST_TEMPLATE.md` to replace the “changelog file” checkbox with a link to the new process and concrete guidance for PR authors.

---

### 3. Describe each step to reproduce the issue or behaviour.

1. Open a new PR in the current trunk — the checklist still refers to the outdated changelog process.  
2. Contributors are unsure whether to create a changelog file or edit `RELEASE_INFO`/`UPGRADE`.  
3. After this PR, the checklist will directly point to the new process and ADR.  

---

### 4. Please link to the relevant issues (if any).

- Internal reference: *Changelog & Release Notes Process* (Confluence)  
- Superseded ADR: [`adr/_superseded/2020-08-03-implement-new-changelog.md`](adr/_superseded/2020-08-03-implement-new-changelog.md)

---

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change *(not applicable – documentation/ADR only)*  
- [x] I have updated the [PR checklist](.github/PULL_REQUEST_TEMPLATE.md) to point to the new ADR  
- [x] I have added a new ADR (`adr/2025-10-28-changelog-release-info-process.md`) describing the new process  
- [x] I have moved the old changelog ADR to `_superseded/`  
- [x] I have read the contribution requirements and fulfilled them

alvarothomas committed:
Move changelog implementation ADR to superseded folder

Move the old changelog process implementation ADR into adr/_superseded/ folder.

alvarothomas committed:
Add Changelog & Release Info Process ADR

This document outlines the new process for managing changelogs and release information, replacing the previous ADR with a curated in-repo model for developer-facing release notes.

alvarothomas committed:
Update changelog ADR to Changelog & Release Info Process

Updated the changelog ADR to reflect a new process for managing changelog and release information, including a curated in-repo model and updated PR guidelines.

alvarothomas committed:
chore(docs): update PR template to reference new release info process

Updated the checklist in the pull request template to include instructions for updating developer-facing release notes and changelog.

mitelg commented on adr/adr/_superseded/2020-08-03-implement-new-changelog.md:
all superseded ADRs have a small text at the beginning why they are superseded and what the successor is

alvarothomas replied:
Thanks, updated it.

mitelg commented on adr/2025-10-28-changelog-release-info-process.md:
```diff
@@ -0,0 +1,45 @@
+2. We adopt a **curated in-repo model**:
+   - `RELEASE_INFO-6.<currentMajor>.md`: developer-facing release notes with per-minor sections.
+   - `UPGRADE-6.<upcomingMajor>.md`: upgrade instructions and breaking changes.
+3. PR authors must add developer-facing entries as part of their PRs:
```
some examples would be nice, how those files then would look like.

alvarothomas replied:
Good point, adding them.

mitelg commented on adr/2025-10-28-changelog-release-info-process.md:
```diff
@@ -0,0 +1,45 @@
+   - Breaking changes → `UPGRADE-6.Y.md`.
+4. The full exhaustive changelog will remain **auto-generated** from GitHub release notes (commit history and merges) and linked from each release for completeness.
+5. The PR template is updated to reference this process and provide checklist guidance.  
+   CI will later validate that either `RELEASE_INFO` or `UPGRADE` was updated when relevant (or explicitly skipped with justification).
```
do we want to force some format how new entries should look like? or just headline and do what ever you want?

keulinho replied:
Not sure what's the payoff of overly formalizing how that should look like would be 

what i would instead like to do is already have the proper real life examples there, e.g. the upgrade file is already there, with knowing that it would not be messed up by auto merged stuff in the future we could clean that up already to provide a good starting point

similarly we should start out with the already filled out release_info file for 6.7.4 once we start it on a PR by PR basis, that way you already have a starting point and no-one needs to start from a blank page

mitelg replied:
It was not meant like we need something like this. I was just curious :+1: 

having already something people can take as example would be very good. as stated already above, some examples are missing

alvarothomas replied:
Added a hint of expected format.

alvarothomas committed:
Update changelog release info process with examples

Added examples for RELEASE_INFO and UPGRADE files, and another bullet for structure and required information for PR authors.

alvarothomas committed:
Mark ADR as superseded and link to new changelog process

This ADR has been replaced by a new document that outlines the updated changelog process.

mitelg commented on adr/2025-10-28-changelog-release-info-process.md:
````diff
@@ -0,0 +1,64 @@
+```
+Example: UPGRADE-6.8.md
+
+## Break: Deprecated SalesChannelContextSwitcher
````
we will only document breaks in that file anyway right? so prefixing every headline with "break" is obsolete, right? :grin:

alvarothomas replied:
Fair enough, that's redundant. Taken out.

alvarothomas committed:
Move new changelog workflow and guidelines

This ADR outlines the decision to implement a new changelog workflow that addresses issues with the current system. It details the context, decision, consequences, and the new workflow for managing changelog entries and upgrade information.

mitelg commented on adr/2025-10-28-changelog-release-info-process.md:
```diff
@@ -0,0 +1,64 @@
+Example: UPGRADE-6.8.md
+
+## Break: Deprecated SalesChannelContextSwitcher
+- **What changed:** The old service is deprecated and will be removed in 6.8.
```
Do we really want to predefine those three bullet points? or would be sentences enough?

alvarothomas replied:
Sentences is enough, those were just suggestions. I've updated them.

alvarothomas committed:
Delete adr/adr directory

Delete extra adr/adr directory

alvarothomas committed:
Merge branch 'trunk' into docs/changelog-process/adr-update

alvarothomas committed:
UPGRADE-6.8.md Examples Update

Updated UPGRADE-#.# file update taking out the 'break' part.

alvarothomas committed:
Update changelog release info process details

Updated changelog process to clarify automation and sources.

MalteJanz commented on adr/2025-10-28-changelog-release-info-process.md:
```diff
@@ -0,0 +1,69 @@
+---
+title: Replace changelog ADR with "Changelog & Release Info Process"
```
Just to double check, have you verified that this markdown file from 6 months ago that is already in our repository is still up to date? Also maybe add a link to it in this ADR / the PR template, as it is better suited as documentation I think than this decision record 🙂 
https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md

MalteJanz commented on adr/_superseded/2020-08-03-implement-new-changelog.md:
Unrelated to this file but I think there is a little more old stuff to remove / superseed:
1. [x] Are [changelog/_template.md](https://github.com/shopware/shopware/blob/trunk/changelog/_template.md) and [changelog/_example.md](https://github.com/shopware/shopware/blob/trunk/changelog/_example.md) still needed for something? Otherwise I would suggest removing those as they can only confuse people.
2. [ ] The `bin/console changelog:create` command should also be removed I think https://github.com/shopware/shopware/blob/0e7efd3b49cdd9137fe191c07884d67a7a804017/src/Core/Framework/Changelog/Command/ChangelogCreateCommand.php#L20-L25
3. [ ] There is still a lot of code for dealing with the old changelog files, can we remove more dead code here? Or is part of it still needed for generating the release changelog? https://github.com/shopware/shopware/tree/0e7efd3b49cdd9137fe191c07884d67a7a804017/src/Core/Framework/Changelog

alvarothomas replied:
Maybe we should take care of these separately but you're right @MalteJanz.
@keulinho can you please lend a hand here?

alvarothomas replied:
I've removed the changelog files, but I'd like this ADR PR to be focused on documentation. For code removal, let's open a separate, cleanup PR.

mitelg replied:
btw, all changelog related classes are internal and can therefore removed directly without deprecation

keulinho replied:
I'll do a seperate PR for that

alvarothomas replied:
Solved in https://github.com/shopware/shopware/pull/13347 I assume.

alvarothomas committed:
Update documentation process and deprecate commands

Updated the documentation process for releases in Shopware, including automated changelog and new release doc process.

alvarothomas committed:
Delete changelog/_template.md

alvarothomas committed:
Delete changelog/_example.md

mitelg commented on delivery-process/documenting-a-release.md:
```diff
@@ -2,14 +2,7 @@
-2. Compliance is automatically enforced, while DevRel and TDMs curate key entries for clarity.
-3. We maintain one single source of truth (SSOT) in GitHub and
-4. Reduce redundancy and align with internal delivery workflows.
+- Related doc: [Documenting a Release in Shopware](../delivery-process/documenting-a-release.md)
```
it refers to itself? :thinking:

alvarothomas replied:
Wrong link, good catch!

pweyck commented on delivery-process/documenting-a-release.md:
```diff
@@ -20,12 +13,21 @@ To have a structured and automated workflow:
+The complete raw changelog for each release is generated automatically from GitHub when a tag is created.  
+It includes every merged PR and commit and is published on the GitHub **Releases** page.  
+This changelog is **not curated** and complements the human-maintained `RELEASE_INFO` and `UPGRADE` files.  
+It’s primarily for internal engineers, support, and partners.  
```
So the changelog will just be put into the releases notes on the github release page? I love this, since it will simplify the release process (no need to push a commit with the changelog before tagging). We can just tag the commit we want to release. Also no merge back required anymore (except for security releases)

keulinho replied:
@pweyck see the example RELEASE_INFO file, on tagging a release it makes sense to add the template for the new version at the top, but yeah merge back should not be needed anymore

pweyck replied:
> That's the idea, what does your team need to enable this?

Nothing special, should be pretty easy to use the gh api. Or you can even just do it on release/tagging

alvarothomas committed:
Change related doc link in release documentation

Updated the related documentation link to the ADR one.

alvarothomas committed:
Merge branch 'trunk' into docs/changelog-process/adr-update

keulinho committed:
Merge branch 'trunk' into docs/changelog-process/adr-update

keulinho committed:
Add example release info file & adjust danger
